src/main/java/com/github/mproberts/rxtools/map/SubjectMap.java

```diff
@@ -331,8 +331,68 @@ public void runIfBound(K key, final Runnable runnable)
         }
 
         if (isBound) {
-            runnable.run();
+            return processNewFaultForKey(key);
         }
+        return Completable.complete();
+    }
+
+    private Completable processNewFaultForKey(final K key)
+    {
+        return Completable.defer(new Callable<CompletableSource>() {
+            @Override
+            public CompletableSource call() throws Exception {
+                return new CompletableSource() {
+                    @Override
+                    public void subscribe(CompletableObserver completableObserver) {
+                        Function<K, Single<V>> faultHandler = _faultHandler;
+
+                        emitFault(key);
+
+                        if (faultHandler != null) {
+                            try {
+                                Single<V> fault = faultHandler.apply(key);
+
+                                fault.doOnSuccess(new Consumer<V>() {
+                                    @Override
+                                    public void accept(V v) throws Exception {
+                                        WeakReference<Processor<V, V>> weakSource = _weakSources.get(key);
+                                        Processor<V, V> processor;
+                                        if (weakSource != null && (processor = weakSource.get()) != null ) {
+                                            processor.onNext(v);
+                                        }
+                                    }
+                                }).toCompletable().subscribe(completableObserver);
+                            } catch (Exception e) {
+                                completableObserver.onError(e);
+                            }
+                        } else {
+                            completableObserver.onComplete();
+                        }
+                    }
+                };
+            }
+        });
+    }
+
+    /**
+     * Re-emits a fault for all bound keys
+     */
+    public Completable faultAllBound()
+    {
+        List<K> keys = new ArrayList<>(_weakSources.size());
+        _readLock.lock();
+
+        try {
+            keys.addAll(_weakSources.keySet());
+        } finally {
+            _readLock.unlock();
+        }
+
+        List<Completable> allFaultCompletables = new ArrayList<>();
+        for (final K key : keys) {
+            allFaultCompletables.add(processNewFaultForKey(key));
+        }
+        return Completable.concat(allFaultCompletables);
```

Comment:
I think `merge()` would be more intuitive. If I make all keys fault, I expect them to run in parallel.

(needs to be specified in the javadoc too)